src/borg/remote.py

```diff
@@ -101,54 +163,97 @@ def serve(self):
                     if self.repository is not None:
                         self.repository.close()
                     else:
-                        os.write(stderr_fd, "Borg {}: Got connection close before repository was opened.\n"
+                        os.write(stderr_fd, 'Borg {}: Got connection close before repository was opened.\n'
                                  .format(__version__).encode())
                     return
                 unpacker.feed(data)
                 for unpacked in unpacker:
-                    if not (isinstance(unpacked, tuple) and len(unpacked) == 4):
+                    if isinstance(unpacked, dict):
+                        dictFormat = True
+                        msgid = unpacked[b'i']
+                        method = unpacked[b'm'].decode('ascii')
```

Comment:
could be just .decode() - would be utf-8 then, but that is a superset of ascii, so no problem.